schemaregistry/rest_service_test.go

```diff
@@ -83,3 +84,75 @@ func TestConfigureTLS(t *testing.T) {
 		t.Errorf("Should work with valid CA, certificate and key, got %s", err)
 	}
 }
+
+func TestNewAuthHeader(t *testing.T) {
+	url, err := url.Parse("mock://")
+	if err != nil {
+		t.Errorf("Should work with empty config, got %s", err)
+	}
+
+	config := &Config{}
+
+	config.BearerAuthCredentialsSource = "STATIC_TOKEN"
+	config.BasicAuthCredentialsSource = "URL"
+
+	_, err = newAuthHeader(url, config)
+	if err == nil {
+		t.Errorf("Should not work with both basic auth source and bearer auth source")
+	}
+
+	// testing bearer auth
+	config.BasicAuthCredentialsSource = ""
+	_, err = newAuthHeader(url, config)
+	if err == nil {
+		t.Errorf("Should not work if bearer auth token is empty")
+	}
+
+	config.BearerAuthToken = "token"
+	headers, err := newAuthHeader(url, config)
+	if err != nil {
+		t.Errorf("Should work with bearer auth token, got %s", err)
+	} else if val, exists := headers["Authorization"]; !exists || len(val) == 0 ||
```

Comment:
can we also check for the presence of these headers?
```
targetSRClusterKey      = "target-sr-cluster"
targetIdentityPoolIDKey = "Confluent-Identity-Pool-Id"
```

Reply:
Those are only required for Cloud SR

Reply:
added it. Thanks @akhileshm1